### DISCUSSION TITLE
Share Operator Implementation

### DISCUSSION THREAD

**Discussion by caillou** [ReactiveX/rxjs#6493] [Q&A] [answered]
From the [documentation](https://github.com/ReactiveX/rxjs/blob/master/src/internal/operators/share.ts#L58):

> [The `share` operator] is an alias for `multicast(() => new Subject()), refCount()`.

This triggered me to look at the implementation, in order to find out how one can create a "simple" operator.

Yet the source code is fairly complex. Form what I see, neither `multicast`, nor `refCount` are used.

What is the reason for this?

**Comment by voliva** [comment, marked as answer]
I think those docs haven't been updated for v7: multicast + refCount are now deprecated and will be removed on v8.

`share` is now an "atomic" multicasting operator.
